src/Drive.php

```diff
@@ -706,6 +722,144 @@ public function getRoles(): array
         return $roles;
     }
 
+    /**
+     * Permanently delete the current drive share
+     * $permissionId will be provided by getPermissionId()
+     * @param string $permissionId
+     * todo after delete drive object contain old permissions
+     *
+     * @throws BadRequestException
+     * @throws ForbiddenException
+     * @throws HttpException
+     * @throws NotFoundException
+     * @throws UnauthorizedException
+     * @throws InvalidResponseException
+     * @throws InternalServerErrorException
+     */
+    public function deletePermission(string $permissionId): bool
+    {
+        try {
+            $this->getDrivesRootApi()->deletePermissionSpaceRoot(
+                $this->getId(),
+                $permissionId
+            );
+        } catch (ApiException $e) {
+            throw ExceptionHelper::getHttpErrorException($e);
+        }
+        $this->updateDriveObject();
+        return true;
+    }
+
+    /**
+     * get Permission id of specific user
+     *
+     * @throws NotFoundException
+     */
+    public function getPermissionId(string $userId): ?string
```

Comment:
1. why limiting this to a specific user? I would like to query all users that have access to the drive. This way I would have to know first a userId and could ask if the user has access. But how to know what userId to ask about?
2. after getting the permissionId I cannot see a way to find out what that permission is allowed to do.
3. according to the example in https://owncloud.dev/libre-graph-api/#/drives.root/ListPermissionsSpaceRoot one user can have multiple permissions

What about having a function called `getPermissions` that returns an array of `DrivePermission`. The `DrivePermission` class can return things like userId and displayname and also the role etc. ?
The setters could be moved ether into that Class or stay inside the `Drive` class and take a complete object as parameter and not only the permission id